File: mobile/library/swift/LogLevel.swift
import Foundation

/// Available logging levels for an Envoy instance.
/// Note that some levels may be compiled out.
@objc
public enum LogLevel: Int {
  case trace = 0
  case debug = 1
  case info = 2
  case warn = 3
  case error = 4
  case critical = 5
  case off = -1

  /// String representation of the log level.
  var stringValue: String {
    switch self {
    case .trace:
      return "trace"
    case .debug:
      return "debug"
    case .info:
      return "info"
    case .warn:
      return "warn"
    case .error:
      return "error"
    case .critical:
      return "critical"
    case .off:
      return "off"
    }
  }
}

File: mobile/library/kotlin/io/envoyproxy/envoymobile/LogLevel.kt
package io.envoyproxy.envoymobile

/**
 * Available logging levels for an Envoy instance. Note some levels may be compiled out.
 *
 * @param level string representation of a given log level.
 * @param levelInt integer representation of a given log level.
 */
enum class LogLevel(internal val level: String, val levelInt: Int) {
  TRACE("trace", 0),
  DEBUG("debug", 1),
  INFO("info", 2),
  WARN("warn", 3),
  ERROR("error", 4),
  CRITICAL("critical", 5),
  OFF("off", 6);

  companion object {
    /** Converts from numeric int to `LogLevel`. */
    fun from(level: Int): LogLevel {
      return when (level) {
        0 -> TRACE
        1 -> DEBUG
        2 -> INFO
        3 -> WARN
        4 -> ERROR
        5 -> CRITICAL
        else -> OFF
      }
    }
  }
}
